File: 151003/Shafarenko/lab2/REST/Entity/Db/News.cs
using REST.Entity.Common;
using System.ComponentModel.DataAnnotations;

namespace REST.Entity.Db
{
    public class News(string title, string content, DateTime created, DateTime modified) : AbstractEntity
    {
        [MinLength(2)]
        public string Title { get; set; } = title;
        [MinLength(4)]
        public string Content { get; set; } = content;
        public DateTime Created { get; set; } = created;
        public DateTime Modified { get; set; } = modified;
        public ICollection<Post> Posts { get; set; } = [];
        public ICollection<Marker> Markers { get; set; } = [];
        public int CreatorId { get; set; }
        public Creator Creator { get; set; } = null!;

        public News() : this(string.Empty, string.Empty, DateTime.Now, DateTime.Now) { }
        public News(int id, int creatorId, string title, string content, DateTime created, DateTime modified)
            : this(title, content, created, modified)
        {
            Id = id;
            CreatorId = creatorId;
            Creator = new() { Id = creatorId };
        }
    }
}

File: 151002/Luschickij/lab2/DC_lab2/src/main/java/com/luschickij/DC_lab/model/entity/News.java
package com.luschickij.DC_lab.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "tbl_news")
public class News {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinColumn(name = "creator_id")
    private Creator creator;

    @Column(name = "title", unique = true, length = 64)
    private String title;

    @Column(name = "content", length = 2048)
    private String content;

    @Column(name = "created")
    private LocalDateTime created;

    @Column(name = "modified")
    private LocalDateTime modified;

    @OneToMany(mappedBy = "news", cascade = CascadeType.ALL)
    private List<Post> posts = new ArrayList<>();
}
